contrib/golang/filters/http/test/golang_integration_test.cc

```diff
@@ -1770,4 +1817,36 @@ TEST_P(GolangIntegrationTest, MissingSecretGoRoutine) {
   testSecrets("missing_secret", "", "404", "/async");
 }
 
+// Set a valid IP address
+TEST_P(GolangIntegrationTest, SetUpstreamOverrideHost) {
+  const std::string host = GetParam() == Network::Address::IpVersion::v4 ? "127.0.0.1" : "[::1]";
+  testUpstreamOverrideHost("200", host, "/test?upstreamOverrideHost=" + host, false, false, false);
+}
+
+// Set a non-IP address,
+TEST_P(GolangIntegrationTest, SetUpstreamOverrideHost_BadHost) {
+  testUpstreamOverrideHost("403", "", "/test?upstreamOverrideHost=badhost", false, true, false);
+}
+
+// Set a invalid IP address
+TEST_P(GolangIntegrationTest, SetUpstreamOverrideHost_InvalidHost) {
+  const std::string host = GetParam() == Network::Address::IpVersion::v4 ? "127.0.0.1" : "[::1]";
+  testUpstreamOverrideHost("200", host, "/test?upstreamOverrideHost=200.0.0.1:8080", false, false,
+                           true);
+}
+
+// Set a invalid IP address with strict mode
+TEST_P(GolangIntegrationTest, SetUpstreamOverrideHost_InvalidHost_Strict) {
+  testUpstreamOverrideHost(
+      "503", "", "/test?upstreamOverrideHost=200.0.0.1:8080&upstreamOverrideHostStrict=true", true,
+      false, true);
+}
+
+// Set a invalid IP address with strict mode and retry
+TEST_P(GolangIntegrationTest, SetUpstreamOverrideHost_InvalidHost_Strict_Retry) {
+  testUpstreamOverrideHost(
+      "503", "", "/test?upstreamOverrideHost=200.0.0.1:8080&upstreamOverrideHostStrict=true", true,
```

Comment:
so, it won't try other host, right? how about non strict mode?
better add some doc about the retry behaviour, thanks.